cmake/AzAutoGen.py

```diff
@@ -24,6 +24,22 @@
 MAX_ERRORS = 100
 errorCount = 0
 
+def ParseCmakeFiles(sourceDir, inputFileList):
+    result = []
+    for cmakeFile in inputFileList:
+        if cmakeFile:
+            sourceCmakeFile = os.path.join(sourceDir, cmakeFile).strip()
+            with open(sourceCmakeFile, 'r') as file:
+                rawCmakeFileContent = file.readlines()
+                for line in rawCmakeFileContent:
+                    cleanLines = line.strip().splitlines() # clean each line by removing line breaker
+                    if len(cleanLines) >= 1 \
+                        and (cleanLines[0].endswith(".xml") \
+                            or cleanLines[0].endswith(".json") \
+                            or cleanLines[0].endswith(".jinja")):
+                        result.append(cleanLines[0])
+    return result
+
```

Comment:
I don't agree with trying to parse contents from a cmake script if it can be avoided.
There are so many gotchas with this parsing, that will cause problems.

For example if the `ALLFILES` variable contained filenames on the same line where the `set` command started.
i.e
```
set(ALLFILES autogen1.xml
    autogen2.xml
)
```

This parsing will fail.

This parsing also fails if the there is a filename on the line with the closing parenthesis
```
set(ALLFILES
    autogen1.xml
    autogen2.xml)
```

In our o3de CLI scripts we are very careful about parsing the .cmake file with a command that sets the `ENABLED_GEMS` variable and even that parsing is vulnerable to issues.
i.e `set( ENABLED_GEM ...)` would fail.

https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/cmake.py#L37-L100

Reply:
good call out, I assumed cmake file has to be in good format
sounds like I have to use regex to parse and get expected file name like
```
([a-zA-Z0-9\/\\]*)\.(xml|json|jinja)
```

Reply:
Any solution should not involve python trying to parse a cmake at all.
Especially not with using regular expressions as CMake can't be parsed completely with regex. pretty much the only tool that can parse CMake script is cmake itself.

I think a good solution to use is have CMake to output the list of the AZCG_INPUTFILES to a plain text file and then have python just read that file.
That constrains the parsing logic to a known format.

I detail that in the comment below
https://github.com/o3de/o3de/pull/8927#discussion_r850684739